### PR TITLE
client-sdk/go: Update validity rules for config identifiers

### PR DESCRIPTION
**Pull request by kostko**

### CODE REVIEW

File: client-sdk/go/config/identifier.go [hunk 1]
```diff
@@ -14,6 +14,8 @@ func ValidateIdentifier(id string) error {
 		return fmt.Errorf("identifier cannot be empty")
 	case len(id) > 64:
 		return fmt.Errorf("identifier must be less than 64 characters long")
+	case id == "default":
+		return fmt.Errorf("identifier cannot be the string 'default'")
 	case !validID.MatchString(id):
 		return fmt.Errorf("identifier must only contain lower-case letters, numbers and _")
```
Comment (peternose) [inline client-sdk/go/config/identifier.go:20]: The error message could also be updated.

File: client-sdk/go/config/identifier.go [hunk 2]
```diff
@@ -5,7 +5,7 @@ import (
 	"regexp"
 )
 
-var validID = regexp.MustCompile(`^[a-z0-9_]+$`)
+var validID = regexp.MustCompile(`^[a-z0-9][a-z0-9_]+$`)
```
Comment (peternose) [inline client-sdk/go/config/identifier.go:8]: Should we use * instead of +?

File: client-sdk/go/config/identifier.go (hunk 1 above)
Comment (kostko) [reply client-sdk/go/config/identifier.go]: Fixed.

File: client-sdk/go/config/identifier.go (hunk 2 above)
Comment (kostko) [reply client-sdk/go/config/identifier.go]: Yeah, fixed and added tests.

**Review by peternose — APPROVED** [review]

**Pull request merged**